pkg/k8s/token.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package k8s
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	authenticationv1 "k8s.io/api/authentication/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	kubeinformers "k8s.io/client-go/informers"
+	storageinformers "k8s.io/client-go/informers/storage/v1"
+	"k8s.io/client-go/kubernetes"
+	storagelisters "k8s.io/client-go/listers/storage/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/kubelet/token"
+)
+
+// TokenClient is a client for Kubernetes Token API
+type TokenClient struct {
+	driverName        string
+	csiDriverInformer storageinformers.CSIDriverInformer
+	csiDriverLister   storagelisters.CSIDriverLister
+	manager           *token.Manager
+}
+
+// NewTokenClient creates a new TokenClient
+// The client will be used to request a token for the preconfigured audiences (--audiences) and expiration time.
+func NewTokenClient(kubeClient kubernetes.Interface, driverName string) *TokenClient {
+	kubeInformerFactory := kubeinformers.NewFilteredSharedInformerFactory(
+		kubeClient,
+		time.Minute*10,
```

Comment:
making this a const and/or adding a comment about why 10 minutes would help.

i.e. should this be configurable like the `audience`?

Reply:
This is the resync period for the informer. The resync plays back all the events held in the informer cache. This is a controller setting for informer cache and is not something we want to expose to the user.  I can make this configurable by the caller similar to https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/c79c919abae3df5ad7c542f67082ec46e05b538b/pkg/k8s/store.go#L57